packages/node-provider/test/unit/node-provider.spec.ts

```diff
@@ -1,7 +1,37 @@
 import NodeProvider from "../../src/node-provider";
+import {
+  NodeMessage,
+  NodeMessageType,
+  NodeProviderStatus
+} from "../../src/types";
 
 describe("NodeProvider", () => {
   it("should instantiate", () => {
     new NodeProvider();
   });
+  it("should connect", async () => {
+    const nodeProvider = new NodeProvider();
+    await nodeProvider.connect();
+
+    expect(nodeProvider.status).toEqual(NodeProviderStatus.CONNECTED);
+  });
+  it("should post a message", () => {
+    const nodeProvider = new NodeProvider();
+
+    nodeProvider.postMessage({
+      messageType: NodeMessageType.INSTALL,
+      requestId: "0",
+      data: null
```

Comment:
I've noticed the `install` type has no expected data payload. Is this accurate?

Reply:
I think we need to split `NodeMesage` into two separate request and response types

Reply:
https://github.com/counterfactual/monorepo/blob/master/packages/cf.js/API_REFERENCE.md#method-install 

Install expects an `appInstanceId` param.

Also I think the split should be elsewhere @ebryn: between messages for "Methods" and for "Events". The former having a defined request-response format and the latter being spontaneous.